### PR TITLE
Removing ref to old google group; adding forum

### PR DESCRIPTION
**Pull request by moxiegirl**
- User google group was shut down
- The email for joining returns a failure about posting
- Adding the new forum

ping @thaJeztah

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -223,16 +223,18 @@ We are always open to suggestions on process improvements, and are always lookin
...
+  <tr>
     <td>Google Groups</td>
     <td>
-      There are two groups.
-      <a href="https://groups.google.com/forum/#!forum/docker-user" target="_blank">Docker-user</a>
-      is for people using Docker containers.
       The <a href="https://groups.google.com/forum/#!forum/docker-dev" target="_blank">docker-dev</a>
       group is for contributors and other people contributing to the Docker
       project.
-      You can join them without an google account by sending an email to e.g. "docker-user+subscribe@googlegroups.com".
```
Comment (thaJeztah) [inline README.md]: guess we can use "docker-dev+subscribe@googlegroups.com" for the dev list?

**Comment by thaJeztah** [comment]
one question, otherwise LGTM

**Comment by moxiegirl** [comment]
@thaJeztah done.

File: README.md [hunk 2]
```diff
@@ -223,16 +223,22 @@ We are always open to suggestions on process improvements, and are always lookin
...
-      The <a href="https://groups.google.com/forum/#!forum/docker-dev" target="_blank">docker-dev</a>
-      group is for contributors and other people contributing to the Docker
-      project.
-      You can join them without an google account by sending an email to e.g. "docker-user+subscribe@googlegroups.com".
-      After receiving the join-request message, you can simply reply to that to confirm the subscribtion.
+      The <a href="https://groups.google.com/forum/#!forum/docker-dev"
+      target="_blank">docker-dev</a> group is for contributors and other people
+      contributing to the Docker project.  You can join this group without an
+      Google account by sending an email to <a
+      href="mailto:docker-user+subscribe@googlegroups.com">docker-user+subscribe@googlegroups.com</a>.
```
Comment (thaJeztah) [inline README.md]: s/docker-user/docker-dev/ :wink:

**Comment by moxiegirl** [comment]
and done... @thaJeztah

**Comment by thaJeztah** [comment]
LGTM \o/

**Pull request merged**